msft-windows/msft-windows-rds-logon-report.ps1

```
@@ -0,0 +1,190 @@
+# script, so treat that as RMM mode even if $RMM was not explicitly passed.
+try {
+    $cmdLineArgs = [Environment]::GetCommandLineArgs()
+    if ($cmdLineArgs | Where-Object { $_ -match '^-NonInteractive$' }) {
+        if ($RMM -ne 1) {
+            Write-Host "Non-interactive PowerShell detected; treating as RMM mode."
+            $RMM = 1
+        }
+    }
+} catch {
+    # If detection itself fails, leave $RMM as-is and proceed.
+}
```

PSScriptAnalyzer (1.25.0)

[warning] Empty catch block is used. Please use Write-Error or throw statements in catch blocks.

(PSAvoidUsingEmptyCatchBlock)